sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup  sys_xtimer Timers
+ * @ingroup   sys
+ * @brief     Provides a high level timer module to register
+ *            timers, get current system time, and let a thread sleep for
+ *            a certain amount of time.
+ *
+ * The implementation takes one low-level timer that is supposed to run at 1MHz
+ * speed and multiplexes it.
+ *
+ * Insertion and removal of timers has O(n) complexity with (n) being the
+ * number of active timers.  The reason for this is that multiplexing is
```

Comment:
> And what happens if there is no memory left for a new timers? *Never* failing is maybe not accurate.

There's no memory allocation within xtimer that could go wrong.
Theoretically, setting an xtimer never fails.

>> >  a failed timer must be dealt with
> is IMO not true, because a failed timer set indicates a critical system failure due to immature application design.
So by mature design, we never hit the "limit". Why do we set it then?

Checking the return value might be cheap. Adapting the logic to do
something more senseful than crash is...

Actually, I'm out. I will not implement the changes needed for limiting
the number of active timers. There are way better ways for getting
realtime timers into RIOT.